src/dataset/infrastructure/mappers/JSDatasetPreviewMapper.ts

```diff
@@ -0,0 +1,30 @@
+import { DatasetPreview as JSDatasetPreview } from '@iqss/dataverse-client-javascript/dist/datasets/domain/models/DatasetPreview'
+import { DatasetPreview } from '../../domain/models/DatasetPreview'
+import {
+  DatasetVersionInfo as JSDatasetVersionInfo,
+  DatasetVersionState as JSDatasetVersionState
+} from '@iqss/dataverse-client-javascript/dist/datasets/domain/models/Dataset'
+import { JSDatasetVersionMapper } from './JSDatasetVersionMapper'
+
+export class JSDatasetPreviewMapper {
+  static toDatasetPreview(jsDatasetPreview: JSDatasetPreview): DatasetPreview {
+    return new DatasetPreview(
+      jsDatasetPreview.persistentId,
+      jsDatasetPreview.title,
+      JSDatasetVersionMapper.toDatasetVersion(
+        jsDatasetPreview.versionId,
+        jsDatasetPreview.versionInfo
+      ),
+      jsDatasetPreview.versionInfo.releaseTime == undefined ? true : false,
```

Comment:
I think this condition is the opposite:
```suggestion
      jsDatasetPreview.versionInfo.releaseTime == undefined ? false : true,
```
The unpublished tag is inverted
![image](https://github.com/IQSS/dataverse-frontend/assets/23359572/a731b177-b584-4da5-8294-c90de16595c6)
![image](https://github.com/IQSS/dataverse-frontend/assets/23359572/c875dcbc-9750-444a-b648-74d29686a756)

Reply:
Since it's a small change I made the commit to approve the PR